core/templates/dev/head/domain/exploration/SolutionObjectFactorySpec.ts

```diff
@@ -84,6 +107,62 @@ describe('Solution object factory', () => {
       });
       expect(solution.getSummary('FractionInput')).toEqual(
         'One solution is "1/6". This is the explanation to the answer.');
+
+      solution.setCorrectAnswer({
+        correct: true
+      });
+      expect(solution.getSummary('LogicProof')).toEqual(
+        'One solution is "true". This is the explanation to the answer.');
+
+      solution.setCorrectAnswer({
+        type: 'real',
+        real: 1,
+        fraction: '',
+        units: []
+      });
+      expect(solution.getSummary('NumberWithUnits')).toEqual(
+        'One solution is "1". This is the explanation to the answer.');
+    });
+
+    it('should get oppia short answer', () => {
+      const interaction = {
+        id: '0',
+        customizationArgs: {
+          choices: {
+            value: 'This is a choice'
+          }
+        }
+      };
+      expect(solution.getOppiaShortAnswerResponseHtml(interaction)).toEqual({
+        prefix: 'One',
+        answer: '<oppia-short-response-0 ' +
+          'answer="&amp;quot;This is a correct answer!&amp;quot;" ' +
+          'choices="&amp;quot;This is a choice&amp;quot;">' +
+          '</oppia-short-response-0>'
+      });
+    });
+
+    it('should change the boolean value of answer exclusivity', () => {
+      solution.setAnswerExclusivity(true);
+      expect(solution.toBackendDict().answer_is_exclusive).toBe(true);
```

Comment:
Do you need to convert it to a dict first? Can't you directly use the value from the solution object itself?

Reply:
Actually I can access using `solution.answerIsExclusive`. Is ok to access the properties of the objects like that? Or should I always use a method that returns the same information?

Reply:
Done.